core-rust/state-manager/src/mempool/pending_transaction_result_cache.rs

```diff
@@ -259,6 +262,23 @@ pub enum AtState {
     },
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct AtStateIntoStateVersionError(pub ());
```

Comment:
what does `(pub ())` do? how does it differ from `{ }`?

Reply:
It does the same thing. This is the same pattern used in `std` (i.e. `pub struct TryFromIntError(pub(crate) ());`).

Here is a comprehensive stackoverflow [thread](https://stackoverflow.com/questions/50162597/what-are-the-differences-between-the-multiple-ways-to-create-zero-sized-structs). Unit-valued tuple struct (current one) is the only way to make it impossible to construct the struct outside it's crate (obviously, this is not the case for this PR as it does have a `pub`). The `{ }` is might mess with macros and generated code (see https://rust-lang.github.io/rfcs/0218-empty-struct-with-braces.html).